crates/wasmtime/src/runtime.rs

```diff
@@ -775,12 +800,35 @@ impl Store {
         self.engine().config.memory_creator.as_ref().map(|x| x as _)
     }
 
-    pub(crate) fn compiler(&self) -> std::cell::Ref<'_, Compiler> {
-        self.inner.compiler.borrow()
+    pub(crate) fn signatures(&self) -> std::cell::Ref<'_, SignatureRegistry> {
```

Comment:
FWIW now that we're doubling down on `Rc` inside of `Store`, I think we should remove these methods. It's too easy, at callsites, to forget these methods are related to `RefCell` and can introduce panics.

Reply:
(removed in favor of direct field access at callsites, that is)

Reply:
Or, alternatively, all accesses to these fields would be scoped inside of methods on this type, where each method doesn't leak borrowed `RefCell`s through returns